bpf/lib/nodeport.h

```diff
@@ -1438,6 +1525,25 @@ nodeport_extract_dsr_v4(struct __ctx_buff *ctx, struct iphdr *ip4,
 		}
 	}
 
+#if defined(IS_BPF_OVERLAY) && \
+	defined(ENABLE_DSR) && DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+	{
+		struct geneve_opt4 gopt;
+		int ret = 0;
+
+		ret = ctx_get_tunnel_opt(ctx, &gopt, sizeof(gopt));
+
+		if (ret > 0) {
+			if (bpf_ntohs(gopt.opt_class) == DSR_GENEVE_OPT_CLASS &&
```

Comment:
```suggestion
			if (gopt.opt_class == bpf_htons(DSR_GENEVE_OPT_CLASS) &&
```

Small trick that allows us to perform the byte-swap at compile-time.